validator/client/validator_metrics.go

```diff
@@ -24,13 +24,14 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 		return nil
 	}
 
-	epoch := slot / params.BeaconConfig().SlotsPerEpoch
-	if epoch == params.BeaconConfig().GenesisEpoch {
-		v.prevBalance = params.BeaconConfig().MaxDepositAmount
+	if (len(v.prevBalances) == 0) {
+		for i := 0; i < len(v.pubkeys); i++ {
+			v.prevBalances = append(v.prevBalances, params.BeaconConfig().MaxDepositAmount);
```

Comment:
this isnt correct, you don't know what the validator balance is. They could have less than or more than the max effective balance(32 ETH)